### DISCUSSION TITLE
Embedding lighter font

### DISCUSSION THREAD

**Discussion by hydroper** [airsdk/Adobe-Runtime-Support#3096] [Q&A] [answered]
(Re-posting from https://forum.starling-framework.org/d/23934-embedding-font-not-really-working)

Following https://airsdk.dev/docs/development/text/embedding-fonts, I've wrote this `[Embed]` meta-data:

```as3
package fortoresseXY.fonts {
    import flash.text.Font;

    public class FortoresseXYFonts {
        ...

        [Embed(
            source = "./notoSans/NotoSans-ExtraLight.ttf",
            fontName = "XNoto Sans ExtraLight",
            fontWeight = "100",
            fontStyle = "normal",
            mimeType = "application/x-font-truetype",
            advancedAntiAliasing = "true",
            embedAsCFF = "false"
        )]
        public static const notoSansExtraLight: Class;

        ...

        public static function register(): void {
            ...
            Font.registerFont(notoSansExtraLight);
            ...
        }
    }
}
```

I've added the `X` prefix to not use my local system font.

In a Feathers UI theme I'm using in my application within `Theme.setTheme(...)`, I've done this:

```as3
override public function getTextFormat(align: String = null): TextFormat {
    const tf: TextFormat = super.getTextFormat(align);
    tf.font = "Noto Sans";
    return tf;
}

private function getHeadingTextFormat(align: String = null): TextFormat {
    const tf: TextFormat = this.getTextFormat(align);
    tf.font = "XNoto Sans ExtraLight";
    tf.size = this.headerFontSize;
    return tf;
}
```

When I remove the `X` prefix from this `TextFormat` used in heading label, the Noto Sans ExtraLight font renders fine. Otherwise, the default AIR font is used.

**Comment by ajwfrost** [comment]
Hi
I can't see the full details from this snippet, is it possible to get a full demo app for it? There are so many different ways in which text fields and fonts can be used...
The above shows a non-CFF font so you would need to be using TextField (rather than TextBlock/FTE) and setting the `TextField.embedFonts` property to true...
thanks

**Reply by hydroper** [reply]
Yeah, I'm using Feathers UI OpenFL SWC for AS3 which uses a `TextField` apparently. I've set `embedFonts = true`.

Here's the project (check the `fortoresseXY.client` directory): https://github.com/hydroper/fortoresseXY

**Reply by hydroper** [reply]
Sorry for sharing the whole project, but it's not too big yet.

[demo.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/14474684/demo.zip)

**Reply by hydroper** [reply, marked as answer]
I've solved the problem as Josh defines an `embedFonts` property in the `Label` control, so I shouldn't use `label.textField.embedFonts`, but just `label.embedFonts`.
